File: solution/day04/Routine.Tests/Fakes/ReindeerFeeder.cs
using FluentAssertions;

namespace Routine.Tests.Fakes;

public class ReindeerFeeder : IReindeerFeeder
{
    private bool _hasBeenFed;
    public void FeedReindeers() => _hasBeenFed = true;
    public void HasBeenFed() => _hasBeenFed.Should().BeTrue();
}

File: exercise/ts/day04/tests/doubles/reindeerFeeder.ts
import { ReindeerFeeder } from '../../src/routine/reindeerFeeder';

export class ReindeerFeederDouble implements ReindeerFeeder {
    private _feedReindeersCalled = false;
    get feedReindeersCalled() {
        return this._feedReindeersCalled;
    }
    feedReindeers() {
        this._feedReindeersCalled = true;
    }
}
